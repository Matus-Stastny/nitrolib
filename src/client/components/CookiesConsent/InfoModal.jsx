// @flow strict
import * as React from "react";
import styled from "styled-components";

import Modal from "client/components/Modal";
import Text from "client/components/Text";
import { Consumer as BrandConsumer } from "client/services/brand/context";

const Container = styled.div``;

const Title = styled.h1``;

const Content = styled.p``;

const tKeys = {
  kayakwhite: __("content.cookies.kayakwhite"),
  kiwicom: __("content.cookies.kiwicom"),
  paragraphs: __("content.cookies.paragraphs"),
};

type Props = {|
  onClose: () => void,
|};

const InfoModal = ({ onClose }: Props) => (
  <Modal onClose={onClose}>
    <Container>
      <Title>
        <Text t={__("content.pages.cookies.title")} />
      </Title>
      <Content>
        <BrandConsumer>
          {brand => <Text t={tKeys[brand.content.legal.cookies.t_key]} html />}
        </BrandConsumer>
      </Content>
    </Container>
  </Modal>
);

export default InfoModal;
