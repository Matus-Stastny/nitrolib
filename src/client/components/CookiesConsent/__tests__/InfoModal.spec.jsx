// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import InfoModal from "../InfoModal";

describe("#CookiesConsent - InfoModal", () => {
  test("render", () => {
    const close = jest.fn();

    const wrapper = shallow(<InfoModal close={close} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("pass close", () => {
    const close = jest.fn();

    const wrapper = shallow(<InfoModal close={close} />);

    expect(wrapper.find("Modal").prop("onClose")).toBe(close);
  });
});
