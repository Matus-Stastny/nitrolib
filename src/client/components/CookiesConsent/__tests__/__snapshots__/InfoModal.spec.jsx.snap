// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CookiesConsent/InfoModal render 1`] = `
<Modal
  onClose={[MockFunction]}
>
  <InfoModal__Container>
    <InfoModal__Title>
      <Text
        html={false}
        t="content.pages.cookies.title"
        values={Object {}}
      />
    </InfoModal__Title>
    <InfoModal__Content>
      <[object Object] />
    </InfoModal__Content>
  </InfoModal__Container>
</Modal>
`;
